pkg/config/config.go

```diff
@@ -4,28 +4,58 @@ import (
 	"bytes"
 	"flag"
 	"fmt"
+	"io"
+	"os"
 	"strings"
 )
 
+type Stream struct {
+	Input  io.Reader
+	Output io.Writer
+	Error  io.Writer
+}
+
 type Config struct {
-	Cache                  string
-	Debug                  bool
-	ExitOnError            bool
-	Files                  []string
-	SchemaLocations        []string
-	SkipTLS                bool
-	SkipKinds              map[string]struct{}
+	Cache                  string   `yaml:"cache" json:"cache"`
+	Debug                  bool     `yaml:"debug" json:"debug"`
+	ExitOnError            bool     `yaml:"exitOnError" json:"exitOnError"`
+	Files                  []string `yaml:"files" json:"files"`
+	Help                   bool     `yaml:"help" json:"help"`
+	IgnoreFilenamePatterns []string `yaml:"ignoreFilenamePatterns" json:"ignoreFilenamePatterns"`
+	IgnoreMissingSchemas   bool     `yaml:"ignoreMissingSchemas" json:"ignoreMissingSchemas"`
+	KubernetesVersion      string   `yaml:"kubernetesVersion" json:"kubernetesVersion"`
+	NumberOfWorkers        int      `yaml:"numberOfWorkers" json:"numberOfWorkers"`
+	OutputFormat           string   `yaml:"output" json:"output"`
 	RejectKinds            map[string]struct{}
-	OutputFormat           string
-	KubernetesVersion      string
-	NumberOfWorkers        int
-	Summary                bool
-	Strict                 bool
-	Verbose                bool
-	IgnoreMissingSchemas   bool
-	IgnoreFilenamePatterns []string
-	Help                   bool
-	Version                bool
+	RejectKindsNG          []string `yaml:"reject" json:"reject"`
+	SchemaLocations        []string `yaml:"schemaLocations" json:"schemaLocations"`
+	SkipKinds              map[string]struct{}
+	SkipKindsNG            []string `yaml:"skip" json:"skip"`
+	SkipTLS                bool     `yaml:"insecureSkipTLSVerify" json:"insecureSkipTLSVerify"`
+	Strict                 bool     `yaml:"strict" json:"strict"`
+	Summary                bool     `yaml:"summary" json:"summary"`
+	Verbose                bool     `yaml:"verbose" json:"verbose"`
+	Version                bool     `yaml:"version" json:"version"`
+	Stream                 *Stream
```

Comment:
I'm wondering if Stream fits better here, or as a separate argument to Validate().

Reply:
Maybe in `cmd/kubeconform` if it is exposed (it should be a package, not main to be reusable).

I will think of a struct and it has something like:

```
type Kubeconform struct {
	Stream *Stream
	Config  *Config
}
```

which represents `Kubeconform` as an application.